### ISSUE TITLE
[Feature Request] Different Autopilot AI Types

### ISSUE THREAD

**Issue by JimbaruJimbuktu** [PapaJoesSoup/BDArmory#551]
Issue Type:  
Feature Request
Summary: 
TL:DR add in a bomber, attack/dive bomber, and fighter classes to aircraft autopilot

Similar to the ground control ai, there could be a slider on the aircraft ai that switches between three classes of planes. The three classes could be fighters, attack/dive bomber, and bomber.
Fighter Class: The fighter class would work as the current ai works. being able to fly in high g, tight maneuvers, and engaging air targets more than anything else.
Attack/Dive bomber Class: The attack class would be somewhere in between a fighter and a bomber. Still able to pull off high G maneuvers, the attack would fly in steep dives to bomb or strafe target positions. Possibly lowering the minimum altitude setting to 50 meters or so to allow torpedo dives to work
Bomber Class: The bomber class would act more strategically, flying at higher altitudes, and not pulling high g maneuvers or extreme rolling maneuvers. Would engage targets flying level, taking into account the type of bomb/missile equipped.
With proper craft design, I think this could complement some of the BDArmory family mods nicely.

**Comment by niteguard12** [comment]
I came for another similar idea of mine!

if the aircraft is a fighter/ attacker, there should be a toggle between agressive and defensive, if aggressive, the plane tries to go to dogfighting range asap, and the default fire rate (the first setting in the weapon manager forgot what it is) will become default 5.

If defensive, the default fire rate is 3 (normal) and tries to stay away from enemy. (If you can, maybe can get closer if behind only)
